### PR TITLE
feat(p/ufmt): Support more formatting verbs

### PR DESCRIPTION
**Pull request by notJoon**
# Description

Support for several new formatting verbs to the `ufmt` package:

`%x`: Outputs byte values as hexadecimal strings. Supports uint8, []uint8, and [32]uint8 types.
`%q`: Outputs quoted strings with proper escaping. Supports string type.
`%T`: Outputs the type of the value. Supports various built-in types using type switching with interface{}.

Particularly in the case of byte slices, previously need to use a loop to convert each element to a string one by one, but now there is no need for that anymore.

### CODE REVIEW

File: examples/gno.land/p/demo/ufmt/ufmt.gno [hunk 1]
```diff
@@ -129,6 +135,70 @@ func Sprintf(format string, args ...interface{}) string {
...
+			case uint8:
+				buf += strconv.FormatUint(uint64(v), 16)
+			case []uint8:
+				for _, b := range v {
+					buf += strconv.FormatUint(uint64(b), 16)
+				}
+			case [32]uint8:
+				for _, b := range v {
+					buf += strconv.FormatUint(uint64(b), 16)
+				}
```
Comment (mvertes) [inline examples/gno.land/p/demo/ufmt/ufmt.gno:149]:
```suggestion
			case []uint8, [32]uint8:
				for _, b := range v {
					buf += strconv.FormatUint(uint64(b), 16)
				}
```

**Review by mvertes — APPROVED** [review]
Looks good to me, except a minor simplification.

**Review by moul — APPROVED** [review]

File: examples/gno.land/p/demo/ufmt/ufmt.gno [hunk 2]
```diff
@@ -129,6 +135,66 @@ func Sprintf(format string, args ...interface{}) string {
...
+			switch v := arg.(type) {
+			case uint8:
+				buf += strconv.FormatUint(uint64(v), 16)
+			case []uint8, [32]uint8:
+				for _, b := range v {
+					buf += strconv.FormatUint(uint64(b), 16)
+				}
+			default:
+				buf += "(unhandled)"
+			}
```
Comment (thehowl) [inline examples/gno.land/p/demo/ufmt/ufmt.gno:148]: this code... isn't supposed to work... https://go.dev/play/p/wzaZ2BdezvX

Comment (thehowl) [reply examples/gno.land/p/demo/ufmt/ufmt.gno:148]: yeah ok, it doesn't in gno, either. that's why the build is failing.

**Review by thehowl — APPROVED** [review]
good, please merge master and fix ci.

**Comment by zivkovicmilos** [comment]
Pinging @notJoon for visibility

File: examples/gno.land/p/demo/ufmt/ufmt.gno (hunk 2 above)
Comment (notJoon) [reply examples/gno.land/p/demo/ufmt/ufmt.gno]: Hmm, it doesn't work anymore. I've just removed that lines.

**Comment by notJoon** [comment]
@zivkovicmilos fixed!

**Pull request merged**